hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -57,6 +69,8 @@ public abstract class SCMCommonPlacementPolicy implements PlacementPolicy {
   private final ConfigurationSource conf;
   private final boolean shouldRemovePeers;
 
+  private Function<ContainerReplica, RID> replicaIdentifierFunction;
```

Comment:
I don't think we need to make this generic for now. It causes a lot of change through many other files to pass in this RID function in, and at the end of the day, it is always `ContainerReplica.getReplicaIndex()`. Perhaps in the future there will be something else, but I cannot think what that might be, so I feel it is likely we will always just have `getReplicaIndex`, saying as the rest of the code deals with containerReplicas only.

Reply:
Done. Changed to Integer adding by adding another constructor